dashboard/app/controllers/activities_controller.rb

```diff
@@ -160,7 +160,7 @@ def track_progress_for_user
         submitted: params[:submitted] == "true",
         level_source_id: @level_source.try(:id),
         level: @level,
-        pairings: pairings
```

Comment:
Note that (tracing through the stack trace (forward and backward)) here we (inefficiently) grab user IDs from `session[:pairings]`, query the DB for the associated Users, and map those Users to their user IDs.